rfcs/0157-more-granular-control.md

```diff
@@ -0,0 +1,111 @@
+---
+feature: more-granular-control
+start-date: 2023-08-02
+author: Animesh Sahu
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+To grant more granular low-level control over the system. Majorly, to decouple various low-level components of the operating system and make everything "opt-in."
+  - This change aims to reduce the disparity observed when using nix inside NixOS and outside NixOS.
+  - By making the infrastructure flexible and allowing components to be explicitly chosen, users can adapt to future breaking changes, such as the introduction of pipewire and wayland, even for init systems and bootloaders.
+
+
+# Motivation
+[motivation]: #motivation
+
+One thing perceived negatively about nix is the continuous addition of abstractions aimed at simplifying the overall system.
+
+The problem becomes significant when working with (or swapping) the lower-level components. For instance:
+
+* The bootloader: No successful implementation has been observed using [ZFSBootMenu](https://docs.zfsbootmenu.org) or performing a direct boot with [Unified Kernel Image](https://animeshz.github.io/site/blogs/demystifying-uefi.html#building-a-uki-for-currently-running-desktop-linux) (using gummiboot, for instance).
+* The init: Some projects ([not-os](https://github.com/cleverca22/not-os) / [nix-processmgmt](https://github.com/svanderburg/nix-processmgmt)) have attempted to use runit / openrc, but most of those projects do not provide sufficient control over many current services present in nixpkgs.
+
+And the list goes on...
+
+While the current configuration does work for most people, and many are content with the existing solutions, a more granular control system would offer tremendous possibilities for customization and make the infrastructure flexible for future shifts.
+
+Instead of deprecating pulseaudio's `sound.*` in favor of `services.pipewire.*`, we should aim to build a flexible API that persists even when new standards emerge, reducing confusion caused by generic names like "sound" getting deprecated.
+
+
+# Detailed design
+[design]: #detailed-design
+
+**Everything is opt-in!**
+
+First and foremost, during image building (whether it's nixos-rebuild or vm/iso generation), the filesystem should be empty, with no packages or files.
+
+We can then assemble basic components by defining choices such as the bootloader, kernel, initramfs generator [1], libc, init, and then the choice of sound & de/wm.
+
+This approach also facilitates the use of NixOS-modules on non-NixOS nix installations.
+
+<sub>[1]: Recently [dracut merged](https://github.com/NixOS/nixpkgs/issues/77858), but there are no clear instructions on how to use that.</sub>
```

Comment:
This is not a paper of philosophy of mathematics. Get rid of this footnote - include it as an example for your text or remove it entirely.